src/dp.rs

```diff
@@ -56,10 +49,40 @@ pub struct DiscreteGaussian {
     sigma: Rational,
 }
 
+impl DifferentialPrivacyDistribution for DiscreteGaussian {}
+
 #[allow(unused_variables)]
 impl DiscreteGaussian {
     /// Creates a new zero-mean Discrete Gaussian distribution with standard deviation `sigma`.
     pub fn new(sigma: Rational) -> Self {
         Self { sigma }
     }
 }
+
+/// Strategy to make aggregate shares differentially private, e.g. by adding noise from a specific
+/// type of distribution instantiated with a given DP budget
+pub trait DifferentialPrivacyStrategy {}
+
+/// A zCDP budget used to create a Discrete Gaussian distribution
+#[allow(dead_code)]
+pub struct ZCdpDiscreteGaussian {
+    budget: ZeroConcentratedDifferentialPrivacyBudget,
+}
+
+impl DifferentialPrivacyStrategy for ZCdpDiscreteGaussian {}
+
+impl ZCdpDiscreteGaussian {
+    /// Creates a new Discrete Gaussian by following Theorem 4 from [[CKS20]]
+    ///
+    /// [CKS20]: https://arxiv.org/pdf/2004.00010.pdf
+    pub fn create_distribution(
+        budget: ZeroConcentratedDifferentialPrivacyBudget,
```

Comment:
nit: this could take a reference instead
```suggestion
        budget: &ZeroConcentratedDifferentialPrivacyBudget,
```

Reply:
Thanks!